ft_sourceanalysis.m

```diff
@@ -821,10 +821,16 @@
 
   if hasbaseline
     % baseline and active are only available together for resampling purposes,
+    % or as a noise covariance for SAM beamforming
     % hence I assume here that there are multiple trials in both
-    baseline.avg   = baseline.avg(datchanindx,:);
-    baseline.cov   = baseline.cov(:,datchanindx,datchanindx);
-    baseline.trial = baseline.trial(:,datchanindx,:);
+    if isfield(baseline, 'avg')
```

Comment:
I think that this can be done in a more principled way, e.g. by looking at the 'covdimord' it can be determined whether the noise covariance has its original trials still present.
Also, from the lower level code, it seems that beamformer_sam assumes the baseline covariance to be a square matrix, thus it will fail if the baseline covariance actually contains repetitions.